# The Powerball lottery and the winning conditions

Powerball is an American lottery game offered by 44 states. It is coordinated by the Multi-State Lottery Association (MUSL). Powerball's minimum advertised jackpot is 40$ million (annuity); Powerball's annuity is paid in 30 graduated installments or winners may choose a lump sum payment instead.

Drawings for Powerball are held every Wednesday and Saturday evening at 10:59 p.m. Eastern Time. Since October 7, 2015, the game has used a 5/69 (white balls) + 1/26 (Powerballs) matrix from which winning numbers are chHalogen, manufactured by Smartplay International of Edgewater Park, New Jersey.osen, resulting in odds of 1 in 292,201,338 of winning a jackpot per play. Each play costs  2,or 3 with the Power Play option.

The minimum Powerball bet is $2. In each game, players select five numbers from a set of 69 white balls and one number from 26 red Powerballs; the red ball number can be the same as one of the white balls. The drawing order of the five white balls is irrelevant; all tickets show the white ball numbers in ascending order. Players cannot use the drawn Powerball to match two of their white numbers, or vice versa. Players can select their own numbers, or have the terminal pseudorandomly select the numbers (called "quick pick", "easy pick", etc.).

The dataset we have here is showing data from October 2015 to June 2018.

In [1]:
import numpy as np
import pandas as pd
import math
from scipy.special import comb

In [2]:
#Calculate how many combinations of 5 numbers can be drawn from 69 unique numbers
combinations_five_denominator = comb(69,5).astype(int)
#Calculate how many ways the correct Powerball number can be chosen from 26 unique numbers
ball_combinations_denominator = comb(26,1).astype(int)
#setting the chances of selecting the correct powerball
ball_chances_numerator = 1
#identifying the number of unique balls numbers
unique_numbers = 69
#the number of draws each time
number_of_draws = 5
#showing the results of both combinations possibilities
combinations_five_denominator, ball_combinations_denominator

(11238513, 26)

In [3]:
def probability(rank_of_price, matching_number_of_balls, boolean_matching_powerball):
    #identifying the combination parameters for the number of balls chosen correctly to be drawn
    #from a sample size, drawn from total unique numbers of the balls
    not_correct_draws = number_of_draws - matching_number_of_balls
    samples_not_drawn = unique_numbers - number_of_draws
    prize_combination = comb(5,matching_number_of_balls).astype(int) * comb(samples_not_drawn, not_correct_draws).astype(int)
    #assumin the basic number of ball chances numerator as if the correct ball is not originally selcted
    ball_chances_numerator = 25
    #the boolean shall be yes or no
    if boolean_matching_powerball == 'yes':
        #in case if yes, it means i need to select the correct powerball
        ball_chances_numerator = 1     
    #the number of chances of getting the correct lottery requirment
    prize_numerator    = ball_chances_numerator * prize_combination
    #the total number of possibilities required for this lottery arrangement
    prize_denominator  = ball_combinations_denominator * combinations_five_denominator
    #making the final number as 1 of all possibilites
    final_probability_numerator = (prize_numerator/prize_numerator).astype(int)
    final_probability_denominator = (prize_denominator//prize_numerator).astype(int)
    print('The probabilty of getting {} prize is {} in {}'.format(rank_of_price, final_probability_numerator, final_probability_denominator))

In [4]:
probability('first', 5, 'yes')
print("."*20)
probability('second', 5, 'no')
print("."*20)
probability('third', 4, 'yes')
print("."*20)
probability('fourth', 4, 'no')
print("."*20)
probability('fifth', 3, 'yes')
print("."*20)
probability('sixth', 3, 'no')
print("."*20)
probability('seventh', 2, 'yes')
print("."*20)
probability('eighth', 1, 'yes')
print("."*20)
probability('ninth', 0, 'yes')

The probabilty of getting first prize is 1 in 292201338
....................
The probabilty of getting second prize is 1 in 11688053
....................
The probabilty of getting third prize is 1 in 913129
....................
The probabilty of getting fourth prize is 1 in 36525
....................
The probabilty of getting fifth prize is 1 in 14494
....................
The probabilty of getting sixth prize is 1 in 579
....................
The probabilty of getting seventh prize is 1 in 701
....................
The probabilty of getting eighth prize is 1 in 91
....................
The probabilty of getting ninth prize is 1 in 38


# Computation Method  Explaination

Calculate how many combinations of 5 numbers can be drawn from 69 unique numbers, Thus, there are 11,238,513 different ways in which 5 numbers can be chosen from a total of 69 unique numbers.

Calculate how many combinations of 1 number can be drawn from 26 unique numbers, Thus, there are 26 different ways 1 number can be chosen from a total of 26 unique numbers.

-----------------------------

Top Prize Level:

Match all five numbers plus the Powerball (1 in 292,201,338 odds)

Step 1: Calculate the number of ways in which 5 numbers can be chosen correctly out of 5 numbers drawn from 69 unique numbers.
This means that there is only 1 way in which 5 numbers out of 5 numbers drawn from a field of 69 numbers can be chosen correctly.
Thus, there is only 1 chance in 11,238,513 of correctly choosing all five numbers drawn in the first portion of PowerBall.

Step 2: Calculate how many ways the correct Powerball number can be chosen from 26 unique numbers.
The chance of correctly choosing the Powerball is simply 1 in 26.

Step 3: Determine the chance of choosing both correctly by multiplying these figures together.

------------------------------

Second Prize Level:
Match all five numbers only (1 in 11,688,053.52 odds)

Step 1: The chance of getting 5 numbers correct out of 5 numbers drawn from 69 unique numbers is 1 in 11,238,513

Step 2: The chance of correctly choosing the Powerball is 1 in 26. Therefore, the chances of incorrectly choosing the Powerball are, conversely, 25 in 26.

Step 3: Determine the chances of choosing 5 out of 5 of 69 correctly and getting the Powerball incorrect by multiplying these figures together.

-----------------------------

Third Prize Level: 

Match four numbers plus the Powerball (1 in 913,129.18 odds)

Step 1: Calculate the number of ways in which 4 numbers can be chosen correctly out of 5 numbers drawn from 69 unique numbers. This means that there are 320 different ways in which 4 numbers out of 5 numbers drawn from a field of 69 numbers can be chosen correctly.
Thus, the chances are 320 in 11,238,513 of correctly choosing 4 out of 5 numbers in the first portion of PowerBall.

Step 2: The chance of correctly choosing the Powerball is simply 1 in 26.

Step 3: Determine the chances of choosing 4 out of 5 of 69 correctly and getting the Powerball correct by multiplying these figures together.

-----------------------------

Fourth Prize Level: 

Match four numbers only (1 in 36,525.17 odds)

Step 1: The chances of getting 4 numbers correct out of 5 numbers drawn from 69 unique numbers are 320 in 11,238,513.

Step 2: The chance of correctly choosing the Powerball is 1 in 26. Therefore, the chances of incorrectly choosing the Powerball are, conversely, 25 in 26.

Step 3: Determine the chances of choosing 4 out of 5 of 69 correctly and getting the Powerball incorrect by multiplying these figures together

-------------------------------

Fifth Prize Level: 

Match three numbers plus the Powerball (1 in 14,494.11 odds)

Step 1: Calculate the number of ways in which 3 numbers can be chosen correctly out of 5 numbers drawn from 69 unique numbers.
This means that there are 20,160 different ways in which 3 numbers out of 5 numbers drawn from a field of 69 numbers can be chosen correctly.
Thus, the chances are 20,160 in 11,238,513 of correctly choosing 3 out of 5 numbers in the first portion of PowerBall.

Step 2: The chance of correctly choosing the Powerball is simply 1 in 26.

Step 3: Determine the chances of choosing 3 out of 5 of 69 correctly and getting the Powerball correct by multiplying these figures together.

---------------------------------

Sixth Prize Level: 

Match three numbers only (1 in 579.76 odds)

Step 1: The chances of getting 3 numbers correct out of 5 numbers drawn from 69 unique numbers are 20,160 in 11,238,513.

Step 2: The chance of correctly choosing the Powerball is 1 in 26. Therefore, the chances of incorrectly choosing the Powerball are, conversely, 25 in 26.

Step 3: Determine the chances of choosing 3 out of 5 of 69 correctly and getting the Powerball incorrect by multiplying these figures together.

---------------------------------

Seventh Prize Level: 

Match two numbers plus the Powerball (1 in 701.33 odds)

Step 1: Calculate the number of ways in which 2 numbers can be chosen correctly out of 5 numbers drawn from 69 unique numbers.
This means that there are 416,640 different ways in which 2 numbers out of 5 numbers drawn from a field of 69 numbers can be chosen correctly.
Thus, the chances are 416,640 in 11,238,513 of correctly choosing 2 out of 5 numbers in the first portion of PowerBall.

Step 2: The chance of correctly choosing the Powerball is simply 1 in 26.

Step 3: Determine the chances of choosing 2 out of 5 of 69 correctly and getting the Powerball correct by multiplying these figures together

-----------------------------------

Eighth Prize Level: 

Match one number plus the Powerball (1 in 91.98 odds)

Step 1: Calculate the number of ways in which 1 number can be chosen correctly out of 5 numbers drawn from 69 unique numbers.
This means that there are 3,176,880 different ways in which 1 number out of 5 numbers drawn from a field of 69 numbers can be chosen correctly.
Thus, the chances are 3,176,880 in 11,238,513 of correctly choosing 1 out of 5 numbers in the first portion of PowerBall.

Step 2: The chance of correctly choosing the Powerball is simply 1 in 26.

Step 3: Determine the chances of choosing 1 out of 5 of 69 correctly and getting the Powerball correct by multiplying these figures together

-----------------------------------

Ninth Prize Level: 

Match the Powerball only (1 in 38.32 odds)

Step 1: Calculate the number of ways in which no numbers are chosen correctly out of 5 numbers drawn from 69 unique numbers.
This means that there are 7,624,512 different ways in which no numbers out of 5 numbers drawn from a field of 69 numbers are chosen correctly.
Thus, the chances are 7,624,512 in 11,238,513 of correctly choosing 0 out of 5 numbers in the first portion of PowerBall

Step 2: The chance of correctly choosing the Powerball is simply 1 in 26.

Step 3: Determine the chances of choosing 0 out of 5 of 69 correctly and getting the Powerball correct by multiplying these figures together